### ISSUE TITLE
can't easily extend sets in option maps

### ISSUE THREAD

**Issue by kkinnear** [kkinnear/zprint#4]
In particular, you can't do this: `(set-options! {:fn-force-nl #{:binding}})`, and have it add to the existing set that is the value for `:fn-force-nl`.  At present it replaces the set that is there, and it should not.  It should operate just like the maps to.  This isn't just for `set-options!`, it is true of any options map -- the command line, lein-zprint, etc.

**Comment by mynomoto** [comment]
I would recommend you to take a look at https://github.com/weavejester/meta-merge because if you make this additive how would you remove things from the set? `meta-merge` uses metadata to config this and at least serve as inspiration for a solution.

**Comment by kkinnear** [comment]
Thanks for the comment.  You make a good point about removing things.  I would like to think that I would have thought of that when I went to fix this bug, but there is no guarantee of that.  I'll give this some more thought -- particularly since I'm thinking that I may need another set or maybe two to handle some of the different formatting issues that have come up recently.

**Comment by kkinnear** [comment]
This is fixed in 0.2.10.

Thanks for the pointer to the`meta-merge`.  It didn't quite fit, but was certainly worth looking at.  I ended up adding a configuration option: `:remove {:fn-force-nl #{...} :fn-gt2-force-nl #{...} :fn-gt3-force-nl #{...}}`  which will remove the elements of the sets from the current configuration.   The remove is processed before any additions.    This seemed to fit best with the normal usage, where people want to remove one thing or add one thing.

I also changed the way you add to sets, so now you don't have to replace the entire set, but anything you put into one of the three sets (`:fn-force-nl`, `:fn-gt2-force-nl`, `:fn-gt3-force-nl`) will be added to the existing set, instead of whatever you give replacing the entire set.
